### PR TITLE
Add connect to redshift action

### PR DESCRIPTION
**Pull request by hunterwerlla**
## Screenshot
Kapture 2020-06-15 at 13 35 28

## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jetbrains-ultimate/src-201/software/aws/toolkits/jetbrains/services/redshift/RedshiftUtils.kt [hunk 1]
```diff
@@ -3,19 +3,39 @@
...
+fun DataSourceRegistry.createDatasource(project: Project, cluster: Cluster) {
+    builder
+        .withJdbcAdditionalProperty(CREDENTIAL_ID_PROPERTY, project.activeCredentialProvider().id)
+        .withJdbcAdditionalProperty(REGION_ID_PROPERTY, project.activeRegion().id)
+        .withUrl(cluster.clusterIdentifier())
+        .withUser(cluster.masterUsername())
+        .withUrl("jdbc:redshift://${cluster.endpoint().address()}:${cluster.endpoint().port()}/${cluster.dbName()}")
+        .commit()
+    // TODO FIX_WHEN_MIN_IS_202 set auth provider ID in builder
+    newDataSources.firstOrNull()?.let {
```
Comment (kiiadi) [inline jetbrains-ultimate/src-201/software/aws/toolkits/jetbrains/services/redshift/RedshiftUtils.kt:34]: what is this magic?

File: resources/resources/software/aws/toolkits/resources/localized_messages.properties [hunk 2]
```diff
@@ -483,6 +483,7 @@ rds.validation.setup_guide=See the RDS guide for IAM authentication
 rds.validation.username=A non-empty username is required
 redshift.auth.aws=AWS Auth
 redshift.cluster_id=Cluster ID
+redshift.connect_aws_credentials=Connect With AWS Credentials
```
Comment (kiiadi) [inline resources/resources/software/aws/toolkits/resources/localized_messages.properties:486]: don't think "with" needs to be title cased.

File: jetbrains-ultimate/src-201/software/aws/toolkits/jetbrains/services/redshift/RedshiftUtils.kt (hunk 1 above)
Comment (hunterwerlla) [reply jetbrains-ultimate/src-201/software/aws/toolkits/jetbrains/services/redshift/RedshiftUtils.kt:34]: So we can't set the authproviderid in the builder (it isn't available). Commit adds it to newDataSources (which is a distinct list from the ones you already have), so we can then pull it back out to set the correct credential ID. I could see if there is a less janky way to do it, but I don't think there is

File: resources/resources/software/aws/toolkits/resources/localized_messages.properties (hunk 2 above)
Comment (hunterwerlla) [reply resources/resources/software/aws/toolkits/resources/localized_messages.properties:486]: Makes sense, changed

File: jetbrains-ultimate/tst-201/software/aws/toolkits/jetbrains/services/redshift/actions/CreateDataSourceActionTest.kt [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+            Cluster.builder()
+                .endpoint { it.address(address).port(port) }
+                .masterUsername(username)
+                .build()
+        )
+        assertThat(registry.newDataSources).hasOnlyOneElementSatisfying {
+            assertThat(it.isTemporary).isFalse()
+            assertThat(it.url).contains(port.toString())
+            assertThat(it.url).contains(address)
+            assertThat(it.url).startsWith("jdbc:redshift://")
```
Comment (rli) [inline jetbrains-ultimate/tst-201/software/aws/toolkits/jetbrains/services/redshift/actions/CreateDataSourceActionTest.kt:41]: why not check that the url completely matches?

Comment (hunterwerlla) [reply jetbrains-ultimate/tst-201/software/aws/toolkits/jetbrains/services/redshift/actions/CreateDataSourceActionTest.kt:41]: True, will change (once my IDE finishes indexing)

**Pull request merged**